src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -120,36 +166,107 @@ public class MyAttribute : System.Attribute
 		instance int32 get_P () cil managed 
 	{
 		// Method begins at RVA 0x2050
-		// Code size 6 (0x6)
+		// Code size 13 (0xd)
 		.maxstack 8
-		IL_0000: call int32 C::'<get_P>g__local|1_0'()
-		IL_0005: ret
+		IL_0000: ldarg.0
+		IL_0001: ldarg.0
+		IL_0002: ldfld int32 C::'<P>k__BackingField'
+		IL_0007: call instance int32 C::nameof(int32)
+		IL_000c: ret
 	} // end of method C::get_P
+	.method public hidebysig 
+		instance int32 nameof (
+			int32 x
+		) cil managed 
+	{
+		// Method begins at RVA 0x205e
+		// Code size 2 (0x2)
+		.maxstack 8
+		IL_0000: ldc.i4.0
+		IL_0001: ret
+	} // end of method C::nameof
 	.method public hidebysig specialname rtspecialname 
 		instance void .ctor () cil managed 
 	{
-		// Method begins at RVA 0x2057
+		// Method begins at RVA 0x2061
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: call instance void [mscorlib]System.Object::.ctor()
+		IL_0006: ret
+	} // end of method C::.ctor
+	// Properties
+	.property instance int32 P()
+	{
+		.get instance int32 C::get_P()
+	}
+} // end of class C
+");
+        }
+
+        [Fact]
+        public void TestNameOfField_NameofIsLocalFunctionInvocation()
```

Comment:
Let's also have a positive test for when `field` is the actual name of a field/property/method.